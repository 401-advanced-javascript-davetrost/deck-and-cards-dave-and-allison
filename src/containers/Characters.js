import React, { Component } from 'react';
import PropTypes from 'prop-types';
import DeckOfComps from './DeckOfComps';
import Card from '../components/Card';
import { getCharacters } from '../services/hey-arnold-api';

export default class Characters extends Component {
  state = {
    characters: []
  }

  componentDidMount() {
    getCharacters({ count: this.props.count })
      .then(characters => {
        this.setState({ characters });
      });
  }

  render() {
    return (
      <DeckOfComps
        title="Characters"
        items={this.state.characters.map(character => ({
          ...character,
          title: character.name,
          content: `${character.name} has ${character._id}`
        }))}
        Component={Card}
      />
    );
  }
  
  static propTypes = {
    count: PropTypes.number
  }


}

