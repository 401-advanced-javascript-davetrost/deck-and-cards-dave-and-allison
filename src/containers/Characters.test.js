import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

jest.mock('../services/hey-arnold-api', () => ({
  getCharacters() {
    return Promise.resolve([
      {
        '_id': '5da237699734fdcb7bef8f51',
        'name': 'Arnold Shortman',
        'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333'
      }
    ]);
  }
}));


describe('Characters component', () => {
  
  it('renders', () => {
    const wrapper = shallow(
      <Characters count={5} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
