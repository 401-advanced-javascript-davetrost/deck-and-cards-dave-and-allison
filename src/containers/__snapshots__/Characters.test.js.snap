// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders 1`] = `
<DeckOfComps
  Component={[Function]}
  items={Array []}
  title="Characters"
/>
`;
